### PR TITLE
Upload image

### LINKED ISSUE #20: 10 - upload image
## Value proposition

**As a**  user
**I want** to add an image of my article
**so that I**  can show the article to the potential buyer

## Description

Image

## Acceptance criteria

- the "verkaufen" page should have an optional upload image option
- the "editieren" page should have an optional upload image option for replacing the current image
- the image should be displayed at the article cards and article details

## Tasks

- [ ] create a feature-branch
- [ ] add an image upload icon to "verkaufen" and "editieren" page
- [ ] create a upload image function by clicking the icon

## Complexity

L

### PR DESCRIPTION
**Pull request by EugenMoser**

### CODE REVIEW

File: components/Article/index.js [hunk 1]
```diff
@@ -12,35 +12,41 @@ function Article({ article }) {
...
+          </StyledListItem>
+        </StyledList>
+        <StyledDetailsLink
+          href={`/articleDetails/${id}`}
+          aria-label="Article Details"
+        >
+          <Svg variant={"details"}></Svg>
+        </StyledDetailsLink>
+      </StyledArticle>
+    </>
```
Comment (JessicaLoers) [inline components/Article/index.js]: - you can delete the react fragments `<></>` because the `<StyledArticle>` component encloses your children.
- _nitpicking little comment:_  you can remove the curly braces if the prop is a string, like you did in line 19. I would recommend you to use a consistent spelling. 

```suggestion
        <StyledList>
          <StyledListItem>
            Größe: <span>{size}</span>
          </StyledListItem>
          <StyledListItem>
            Geschlecht:<span>{gender}</span>
          </StyledListItem>
          <StyledListItem>
            Preis (Euro): <span>{price}</span>
          </StyledListItem>
        </StyledList>
        <StyledDetailsLink
          href={`/articleDetails/${id}`}
          aria-label="Article Details"
        >
          <Svg variant="details" />
        </StyledDetailsLink>
      </StyledArticle>
  
```

File: components/Article/index.js [hunk 2]
```diff
@@ -12,35 +12,41 @@ function Article({ article }) {
...
+            src={
+              image === ""
+                ? "https://res.cloudinary.com/depezzq0e/image/upload/v1670228543/placeholder-image_j7hueu.jpg"
+                : image
+            }
+            alt={`Ein Bild von ${name}`}
+            fill
+          />
+        </StyledArticleImageContainer>
+        <h3>{name}</h3>
```
Comment (JessicaLoers) [inline components/Article/index.js]:
```suggestion
      <StyledArticle>
        <StyledArticleImageContainer>
          <StyledImage
            variant="article"
            src={
              image === ""
                ? "https://res.cloudinary.com/depezzq0e/image/upload/v1670228543/placeholder-image_j7hueu.jpg"
                : image
            }
            alt={`Ein Bild von ${name}`}
            fill
          />
        </StyledArticleImageContainer>
        <h3>{name}</h3>
```

File: pages/_app.js [hunk 3]
```diff
@@ -12,15 +13,17 @@ function MyApp({ Component, pageProps }) {
...
-          getArticleById={getArticleById}
-        />
-      </Layout>
+      <CloudinaryContext cloudName="depezzq0e">
+        <GlobalStyles />
+        <Layout>
+          <Component
+            {...pageProps}
+            articles={articles}
+            setArticles={setArticles}
```
Comment (JessicaLoers) [inline pages/_app.js]: Actually this is not part of the PR but I highly recommend to never pass the state-setter (callback function e.g. setArticles) to your components 🚫
➡️ Rather place the `addArticle function` from `articleSell.js` and `articleDetails/[id]/index.js` ...  and pass this function  as a prop `onAddArticle={addArticle}`

File: pages/myArticles.js [hunk 4]
```diff
@@ -15,11 +17,22 @@ function MyArticles({ articles }) {
...
+        <h2>
+          Du hast keine Artikel zum Verkauf.{" "}
+          <StyledLink
+            href="articleSell"
+            variant="linkToSell"
+          >
+            Hier kannst du deine Artikel erstellen
+          </StyledLink>
+        </h2>
+      )}
```
Comment (JessicaLoers) [inline pages/myArticles.js]: Just a nitpicking suggestion 🤓

```suggestion
      {authorArticles.length  ? (
        <ArticleList
          articles={authorArticles}
          secondHeadline="Das sind deine Artikel:"
        />
      ) : (
        <h2>
          Du hast keine Artikel zum Verkauf.
          <StyledLink
            href="articleSell"
            variant="linkToSell"
          >
            Hier kannst du deine Artikel erstellen
          </StyledLink>
        </h2>
      )}
```

**Review by JessicaLoers — COMMENTED** [review]
Hi @EugenMoser 
**very well done** 🚀

The cloudinary related codes looks good to me - even though I couldn't test it in the preview. 
Please check the Environment Variable on vercel 🙂

I noticed that you pass the set-setter function as a property - this works, but can lead to mean bugs.  Please see my comments on how you can easily solve this.

File: pages/myArticles.js (hunk 4 above)
Comment (EugenMoser) [reply pages/myArticles.js]: thanks Jessica. As always top feedback

**Comment by JulianHrt** [comment]
Hi Eugen, now everything is fine and your code is clear and good readable.
